### ISSUE TITLE
Inconsistency in the treatment of non-integer pk in (foreign key) resource URI

### ISSUE THREAD

**Issue by ksze** [django-tastypie/django-tastypie#1133]
There appears to be some inconsistency in the treatment of non-integer pk in resource URI.

Using a combination of:
-   `Django 1.6.2`
-   `django-tastypie 0.11.0`
-   `Backbone.js 1.0.0`

For example, if I have two Django models `SubCategory` and `CollectionItem`:

``` python
from django.db import models
class SubCategory(models.Model):
    name = models.CharField(max_length=200, primary_key=True, verbose_name='Name')

class CollectionItem(models.Model):
    # some fields, then ...
    name = models.CharField()
    sub_category = models.ForeignKey(SubCategory, blank=True, null=True, related_name='collection_items')
```

Then, the corresponding tastypie resource definitions:

``` python
from tastypie import fields
from tastypie.resources import ModelResource

from myapp.models import SubCategory, CollectionItem

class SubCategoryResource(ModelResource):
    class Meta:
        queryset = SubCategory.objects.all().order_by('ordering')

class CollectionItemResource(ModelResource)
    sub_category=fields.ToOneField(SubCategoryResource, "sub_category", full=True, null=True)
    class Meta:
        queryset = CollectionItem.objects.all()
```

If I fetch a CollectionItem in my client, let's say with Backbone.js:

``` JavaScript
var collection_item = new CollectionItem({id: 123});
collection_item.fetch();
```

The response would be like: 

``` JSON
{
    "id": 123,
    "name": "Aston Martin DB5",
    "sub_category": {
        "name": "Vintage Sportscar",
        "resource_uri": "/api/v1/subcategory/Vintage%20Sportscar"
    },
    "resource_uri": "/api/v1/collectionitem/123/"
}
```

Now, we know that we can change a ToOneField (i.e. a foreign key) just by setting it to the proper resource URI. For example:

``` JavaScript
collection_item.set({
    name: 'Aston Martin DBS',
    sub_category: '/api/v1/subcategory/Modern%20Sportscar', // just a URI is sufficient
});
collection_item.save(); // this will send a HTTP PUT request
```

This is where the problem comes in. Tastypie tries to resolve `/api/v1/subcategory/Modern%20Sportscar` to the `SubCategory` object, but fails because somewhere deep in `tastypie.fields.RelatedField.resource_from_uri()`, `django.core.urlresolvers.resolve()` is used and the `pk` of the `SubCategory` object is naïvely resolved as `Modern%20Sportscar` instead of `Modern Sportscar`, so an `ObjectDoesNotExist` exception is raised, even though the `Modern Sportscar` SubCategory does exist; and the request fails with an HTTP 400.

In other words, when I fetch a resource from tastypie, the resource URIs are URI-encoded, but when I save a resource to tastypie, tastypie expects resource URIs to _not_ be URI-encoded. I feel that resource URIs not being URI-encoded would be incorrect, but I might be wrong.
